regolith/helpers/l_progressreporthelper.py

```diff
@@ -0,0 +1,172 @@
+            collaborators = None
+            if p.get('collaborators'):
+                collaborators = ', '.join(p.get('collaborators'))
+            print(f"        group_members: {grp_members}")
+            print(f"        collaborators: {collaborators}")
+            print("    milestones:")
+            for m in p.get('milestones'):
+                print(f"        {m.get('name')}")
+                print(f"            status: {m.get('status')}, due_date: {m.get('due_date')}")
+
+        else:
+            print(f"{p.get('_id')}")
+            if p.get("deliverable"):
+                print(
+                    f"    status: {p.get('status')}, begin_date: {p.get('begin_date')}, due_date: {p.get('deliverable').get('due_date')}")
+            else:
+                print(
+                    f"    status: {p.get('status')}, begin_date: {p.get('begin_date')}, due_date: {p.get('due_date')}")
+
+class ProgressReportHelper(SoutHelperBase):
+    """Helper for listing upcoming (and past) projectum milestones.
+
+       Projecta are small bite-sized project quanta that typically will result in
+       one manuscript.
+    """
+    # btype must be the same as helper target in helper.py
+    btype = HELPER_TARGET
+    needed_dbs = [f'{TARGET_COLL}']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        if "groups" in self.needed_dbs:
+            rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        try:
+            if not rc.database:
+                rc.database = rc.databases[0]["name"]
+        except:
+            pass
+        colls = [
+            sorted(
+                all_docs_from_collection(rc.client, collname), key=_id_key
+            )
+            for collname in self.needed_dbs
+        ]
+        for db, coll in zip(self.needed_dbs, colls):
+            gtx[db] = coll
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+    def sout(self):
+        rc = self.rc
+        if rc.filter:
+            collection = key_value_pair_filter(self.gtx["projecta"], rc.filter)
+        else:
+            collection = self.gtx["projecta"]
+
+        projecta = []
+        for projectum in collection:
+            if (not rc.lead) and (not rc.stati):
+                projecta.append(projectum)
+                continue
+            if rc.lead and projectum.get('lead') != rc.lead:
+                continue
+            if rc.stati and projectum.get('status') not in rc.stati:
+                continue
+            projecta.append(projectum)
+
+        selected_projecta = []
+        for p in projecta:
+            if p.get('status') != "proposed":
+                continue
+            selected_projecta.append(p)
+        if selected_projecta != []:
```

Comment:
These are more readable if you put simply `if selected_projecta:`  

The loop over projecta is probably small so your construction is probably fine but it could also be more readable if you loop over the projecta collection only once and then collect `proposed_projecta`, `started_projecta` and so on, then do your print statements.  Also, this allows you to do a trick to make the code more easily maintained, bbecause you can make a list of the allowed stati (statuses) that you can pull from schema and then loop over that in such a way that if someone adds a new status in the future (this happens!) your code will automatically build that.

Reply:
Fixed.